### PR TITLE
Listens for mentions and calls

### PR DESCRIPTION
**Pull request by AdamT20054**
Listens for mentions, option to make custom commands based on a given input

### CODE REVIEW

**Review by rssnyder — CHANGES_REQUESTED** [review]
looks good, made a few requests for the function name and comments.

File: stock.go [hunk 1]
```diff
@@ -78,10 +78,29 @@ func NewCrypto(ticker string, token string, name string, nickname bool, color bo
 func (s *Stock) Shutdown() {
 	s.close <- 1
 }
+// THIS FUNCTION IS THE SAME FOR STOCK AND CRYPTO, IF YOU WANT DIFFERENT RESPONSES, MAKE A DIFFEREN'T FUNCTION FOR EACH AND CHANGE THE LINES: 142 AND 340 TO THEIR NEW FUNCTION NAME
+func messageCreate(s *discordgo.Session, m *discordgo.MessageCreate) {
```
Comment (rssnyder) [inline stock.go:82]:
```suggestion

// sendMessage responds to users who @ the bot
func sendMessage(s *discordgo.Session, m *discordgo.MessageCreate) {
```

File: stock.go [hunk 2]
```diff
@@ -78,10 +78,29 @@ func NewCrypto(ticker string, token string, name string, nickname bool, color bo
...
+    // the xxxxx should be what the bot responds to, with the 'response here' being the response
+    if m.Content == "xxxxx" {
+        s.ChannelMessageSend(m.ChannelID, "Response here*")
+    }
+}
 
 func (s *Stock) watchStockPrice() {
 	var exRate float64
 
+
```
Comment (rssnyder) [inline stock.go:103]:
```suggestion
```

File: stock.go [hunk 3]
```diff
@@ -119,7 +138,8 @@ func (s *Stock) watchStockPrice() {
 			exRate = exData.QuoteSummary.Results[0].Price.RegularMarketPrice.Raw
 		}
 	}
-
+	
+    	dg.AddHandler(messageCreate)
```
Comment (rssnyder) [inline stock.go:142]:
```suggestion
    	dg.AddHandler(sendMessage)
```

File: stock.go [hunk 4]
```diff
@@ -316,7 +336,8 @@ func (s *Stock) watchCryptoPrice() {
...
```
Comment (rssnyder) [inline stock.go:340]:
```suggestion
    	dg.AddHandler(sendMessage)
```

File: stock.go [hunk 5]
```diff
@@ -78,10 +78,29 @@ func NewCrypto(ticker string, token string, name string, nickname bool, color bo
 func (s *Stock) Shutdown() {
 	s.close <- 1
 }
+// THIS FUNCTION IS THE SAME FOR STOCK AND CRYPTO, IF YOU WANT DIFFERENT RESPONSES, MAKE A DIFFEREN'T FUNCTION FOR EACH AND CHANGE THE LINES: 142 AND 340 TO THEIR NEW FUNCTION NAME
+func messageCreate(s *discordgo.Session, m *discordgo.MessageCreate) {
+    //ignores itself
```
Comment (rssnyder) [inline stock.go:83]:
```suggestion
    // ignores itself
```

File: stock.go [hunk 6]
```diff
@@ -78,10 +78,29 @@ func NewCrypto(ticker string, token string, name string, nickname bool, color bo
...
+    if m.Author.ID == s.State.User.ID {
+        return
+    }
+    //listens for mention
```
Comment (rssnyder) [inline stock.go:87]:
```suggestion

    // listens for mention
```

File: stock.go [hunk 7]
```diff
@@ -78,10 +78,29 @@ func NewCrypto(ticker string, token string, name string, nickname bool, color bo
...
+    for _, user := range m.Mentions {
+        if user.ID == s.State.User.ID {
+            s.ChannelMessageSend(m.ChannelID, "Bot  online;") 
+        }
+    }
+        
+    // the xxxxx should be what the bot responds to, with the 'response here' being the response
+    if m.Content == "xxxxx" {
+        s.ChannelMessageSend(m.ChannelID, "Response here*")
+    }
```
Comment (rssnyder) [inline stock.go:97]: Lets remove this for now if we dont know what to put. Can add back later if we want.

Comment (AdamT20054) [reply stock.go:97]: Yea thats fine, was just a 'format' for anyone wanting to add it. Ill // it out?

**Comment by AdamT20054** [comment]
should be ready to go

File: stock.go [hunk 8]
```diff
@@ -79,6 +79,27 @@ func (s *Stock) Shutdown() {
...
+    for _, user := range m.Mentions {
+        if user.ID == s.State.User.ID {
+            s.ChannelMessageSend(m.ChannelID, "Bot  online;") 
+        }
+    }
+        
+    // the xxxxx should be what the bot responds to, with the 'response here' being the response
+    // if m.Content == "xxxxx" {
+    //    s.ChannelMessageSend(m.ChannelID, "Response here*")
+    //}
```
Comment (rssnyder) [inline stock.go:100]: Can we remove this completely? Not a big fan of commented out code. 

```suggestion
```

Comment (AdamT20054) [reply stock.go:100]: Yea thats fine, might be worth having something about it somewhere in the readme though?

**Comment by AdamT20054** [comment]
NOW it should be all good?

File: stock.go [hunk 9]
```diff
@@ -79,6 +79,23 @@ func (s *Stock) Shutdown() {
...
+
+    // ignores itself
+    if m.Author.ID == s.State.User.ID {
+        return
+    }
+
+    // listens for mention
+    for _, user := range m.Mentions {
+        if user.ID == s.State.User.ID {
+            s.ChannelMessageSend(m.ChannelID, "Bot  online;")
```
Comment (rssnyder) [inline stock.go:93]:
```suggestion
            _, err := s.ChannelMessageSend(m.ChannelID, "Bot  online;")
            if err != nil {
                fmt.Println("Error sending message reply: ", err)
            }
```

**Comment by AdamT20054** [comment]
Ok, nowww it should be good

**Comment by rssnyder** [comment]
I've decided that I don't want the bot to have interactions with users, and just be a dynamic display, so I am going to close this PR for now.

Thanks for the idea tho!

**Pull request closed without merging**